coriolis/tests/osmorphing/test_windows.py

```diff
@@ -0,0 +1,860 @@
+            "HKLM\\%s" % mock_uuid4.return_value,
+            "%sWindows\\System32\\config\\SYSTEM" % self.os_root_dir)
+
+        mock_set_service_start_mode.assert_called_once_with(
+            str(mock_uuid4.return_value), windows.CLOUDBASEINIT_SERVICE_NAME,
+            windows.SERVICE_START_DISABLED)
+        mock_unload_registry_hive.assert_called_once_with(
+            "HKLM\\%s" % mock_uuid4.return_value)
+
+    def test__check_cloudbase_init_exists(self):
+        self.conn.exec_ps_command.return_value = "True"
+        result = self.morphing_tools._check_cloudbase_init_exists(
+            mock.sentinel.key_name)
+
+        self.conn.exec_ps_command.assert_called_once_with(
+            "Test-Path %s" % (windows.SERVICE_PATH_FORMAT % (
+                mock.sentinel.key_name, windows.CLOUDBASEINIT_SERVICE_NAME)))
+
+        self.assertTrue(result)
+
+    @mock.patch.object(
+        windows.BaseWindowsMorphingTools, '_set_service_start_mode'
+    )
+    def test__setup_existing_cbslinit_service(
+            self, mock_set_service_start_mode):
+        self.morphing_tools._setup_existing_cbslinit_service(
+            mock.sentinel.key_name, mock.sentinel.image_path)
+
+        reg_service_path = windows.SERVICE_PATH_FORMAT % (
+            mock.sentinel.key_name, windows.CLOUDBASEINIT_SERVICE_NAME)
+
+        self.conn.exec_ps_command.assert_has_calls([
+            mock.call(
+                "Set-ItemProperty -Path '%s' -Name 'ImagePath' -Value '%s' "
+                "-Force" % (reg_service_path, mock.sentinel.image_path),
+            ),
+            mock.call(
+                "Set-ItemProperty -Path '%s' -Name 'ObjectName' "
+                "-Value 'LocalSystem' -Force" % reg_service_path
+            ),
+        ])
+
+        mock_set_service_start_mode.assert_called_once_with(
+            mock.sentinel.key_name, windows.CLOUDBASEINIT_SERVICE_NAME,
+            windows.SERVICE_START_AUTO)
+
+    def test__get_cbslinit_base_dir(self):
+        result = self.morphing_tools._get_cbslinit_base_dir()
+
+        self.assertEqual(result, "%sCloudbase-Init" % self.os_root_dir)
+
+    def test__get_cbslinit_scripts_dir(self):
+        result = self.morphing_tools._get_cbslinit_scripts_dir('C:')
+
+        self.assertEqual(result, "C:\\LocalScripts")
+
+    @mock.patch.object(windows.utils, 'write_winrm_file')
+    def test__write_local_script(self, mock_write_winrm_file):
+        script_path = "/script/path/script.sh"
+
+        with mock.patch('builtins.open', mock.mock_open()) as mock_open:
+            self.morphing_tools._write_local_script(
+                'C:', script_path, priority=50)
+            mock_open.assert_called_once_with(script_path, 'r')
+            mock_write_winrm_file.assert_called_once_with(
+                self.conn,
+                'C:\\LocalScripts\\50-script.sh',
+                '')
+
+    @mock.patch.object(windows.utils, 'get_resources_bin_dir')
```

Comment:
You don't need to mock `get_resources_bin_dir`.

You can directly use `windows.utils.get_resources_bin_dir()` in the unit test assertions, since it returns a string with a path.

Reply:
Gotcha!